src/flowcean/transforms/match_sampling_rate.py

```diff
@@ -13,99 +14,187 @@ class MatchSamplingRate(Transform):
-    present in the DataFrame. To remove it use the `select` transform.
     """
 
     def __init__(
         self,
-        reference_timestamps: str,
-        feature_columns_with_timestamps: dict[str, str],
+        reference_feature_name: str,
+        feature_interpolation_map: dict[str, str],
     ) -> None:
-        """Initialize the MatchSamplingRate transform.
+        """Initialize the transform.
 
         Args:
-            reference_timestamps: Timestamps of the reference feature.
-            feature_columns_with_timestamps: Names of the features that are
-                getting interpolated with their respective original timestamp
-                feature names.
+            reference_feature_name: Reference timeseries feature.
+            feature_interpolation_map: Key-value pairs of the timeseries
+                features that are targeted in interpolation columns and the
+                interpolation method to use. At the moment, the interpolation
+                method can only be 'linear'.
         """
-        self.reference_timestamps = reference_timestamps
-        self.feature_columns_with_timestamps = feature_columns_with_timestamps
+        self.reference_feature_name = reference_feature_name
+        self.feature_interpolation_map = feature_interpolation_map
 
+    @override
     def transform(self, data: pl.DataFrame) -> pl.DataFrame:
-        logger.debug("Matching sampling rate of time series.")
-
-        for i in range(len(data)):
-            reference_timestamps = data[self.reference_timestamps][i]
-            for (
-                feature,
-                timestamp,
-            ) in self.feature_columns_with_timestamps.items():
-                timestamps = data[timestamp][i].to_numpy()
-                feature_data = data[feature][i].to_numpy()
-                resampled_timeseries = interp(
-                    reference_timestamps,
-                    timestamps,
-                    feature_data,
-                ).tolist()
-                data = (
-                    data.lazy()
-                    .with_row_index()
-                    .with_columns(
-                        pl.when(pl.col("index") == i)
-                        .then(resampled_timeseries)
-                        .otherwise(pl.col(feature))
-                        .alias(feature)
-                    )
-                    .drop("index")
-                    .collect()
+        # preserve all constant columns that are not timeseries data
+        non_timeseries_features = data.select(
+            pl.exclude(
+                *self.feature_interpolation_map.keys(),
+                self.reference_feature_name,
+            )
+        )
+        debug_msg = (
+            "Interpolating timeseries features: "
+            + str(self.feature_interpolation_map.keys())
+            + " using the timestamps of the reference feature "
+            + self.reference_feature_name
+        )
```

Comment:
You can use format strings for that. something like
```python
f"Interpolating timeseries features: {self.feature_interpolation_map.keys()} using the timestamps of the reference feature {self.reference_feature_name}"

```

Reply:
switched to f-strings.